app/components/Settings/OptionsDropdown/OptionsDropdown.react.js

```diff
@@ -1,61 +1,46 @@
-        this.renderElasticsearchDocs = this.renderElasticsearchDocs.bind(this);
-    }
 
     static propTypes = {
         // See type of react-select's Select `value` property
         selectedTable: PropTypes.any,
         selectedIndex: PropTypes.any,
-
         tablesRequest: PropTypes.object,
         setTable: PropTypes.func,
         elasticsearchMappingsRequest: PropTypes.object,
         setIndex: PropTypes.func
     }
 
+    /**
+     * Options Dropdown is an options drop down
+     * @param {object} props  - Component Properties
+     * @param {object} props.selectedTable - The selected table
+     * @param {object} props.tablesRequest - The Requested Table
+     * @param {function} props.setTable - The set table function
+     * @param {object} props.elasticsearchMappingsRequest - The ES Mapping Request
+     * @param {object} props.selectedTable - The selected table
+     * @param {object} props.selectedIndex - The Selected Index
+     * @param {function} props.setIndex - The Set the index
+     */
+    constructor(props) {
+        super(props);
+        this.renderSQLOptions = this.renderSQLOptions.bind(this);
+        this.renderElasticsearchIndecies = this.renderElasticsearchIndecies.bind(this);
+        this.renderElasticsearchDocs = this.renderElasticsearchDocs.bind(this);
+    }
+
     renderSQLOptions() {
         const {selectedTable, tablesRequest, setTable} = this.props;
-        if (!tablesRequest.status) {
-            return null;
-        } else if (tablesRequest.status === 'loading') {
-            return <div>{'Loading tables'}</div>;
-        } else if (tablesRequest.status > 300) {
-            // TODO - Make this prettier.
-            return (
-                <div>
-                    <div>{'Hm.. there was an error loading up your tables'}</div>
-                    <div style={{color: 'red'}}>{JSON.stringify(tablesRequest)}</div>
-                </div>
-            );
-        } else if (tablesRequest.status === 200) {
-            const tablesList = flatten(tablesRequest.content);
-            if (tablesList.length === 0) {
-                return <div>{'No tables found'}</div>;
-            }
-            return (
-                <div className={'dropdown'}
-                    id="test-table-dropdown"
-                >
-                    <Select
-                        options={tablesList.map(t => ({label: t, value: t}))}
-                        value={selectedTable}
-                        searchable={false}
-                        onChange={option => {
-                            setTable(option.value);
-                        }}
-                    />
-                </div>
-            );
-        }
+
+        return (<SQLOptions selectedTable={selectedTable}
+                            tablesRequest={tablesRequest}
+                            setTable={setTable}
+        />);
     }
 
     renderElasticsearchIndecies() {
```

Comment:
Since this code has moved to `<ESIndicesOptions>`, let's get rid of this method.